processor/src/main/java/org/mapstruct/ap/processor/MethodRetrievalProcessor.java

```diff
@@ -80,19 +93,60 @@ public int getPriority() {
         return 1;
     }
 
+    private List<SourceMethod> retrievePrototypeMethods(TypeMirror typeMirror, MapperConfig mapperConfig) {
+        if ( typeMirror == null || typeMirror.getKind() == TypeKind.VOID ) {
+            return Collections.emptyList();
+        }
+
+        TypeElement typeElement = asTypeElement( typeMirror );
+        List<SourceMethod> methods = new ArrayList<SourceMethod>();
+        for ( ExecutableElement executable : getAllEnclosedExecutableElements( elementUtils, typeElement ) ) {
+
+            ExecutableType methodType = typeFactory.getMethodType( typeElement, executable );
+            List<Parameter> parameters = typeFactory.getParameters( methodType, executable );
+            boolean containsTargetTypeParameter = SourceMethod.containsTargetTypeParameter( parameters );
+
+            // prototype methods don't have prototypes themselves
```

Comment:
> I think that this will be a prime use case for me.

Out of interest, why would you prefer to declare prototype methods (= methods to be used as configuration source) in config classes rather than actual mappers?

> I also don't understand why the return type per-se should be abstract.

No, it's not that prototype methods in config classes *have* to declare an abstract return type. What I mean is: I would declare prototype methods in actual mappers whenever it's possible and feasible. I would only declare them in config classes in two cases:

* The config I wish to make inheritable refers to an abstract return type. In this case I cannot declare that method on a mapper, as no implemention can be generated for this method. Thus I'd use a config class
* I don't want to make the prototype method usable for clients of the mapper class (e.g. if I want to achieve that only mappings for actual sub-classes in a hierarchy are invoked). In that case I'd put it into the config class to "hide" it from mapper users.

Otherwise I'd prefer methods on actual mappers for the sake of comprehensability.

> In the very near future, The method to which it applies is a forged method

I'm not sure what you mean. Could you give an example for this case? A user can declare (and configure) e.g. an element mapping method which then is used by an iterable mapping method. Why would they declare that method on a config class rather than a mapper?